frappe/email/receive.py

```diff
@@ -149,37 +153,86 @@ def get_messages(self):
 			else:
 				self.pop.quit()
 
-		return self.latest_messages
+		out = { "latest_messages": self.latest_messages }
+		if self.settings.use_imap:
+			out.update({
+				"uid_list": uid_list,
+				"seen_status": self.seen_status,
+				"unique_id_list": self.unique_id_list
+			})
+
+		return out
 
 	def get_new_mails(self):
 		"""Return list of new mails"""
 		if cint(self.settings.use_imap):
-			self.imap.select("Inbox")
-			response, message = self.imap.uid('search', None, "UNSEEN")
+			self.check_imap_uidvalidity()
+
+			self.imap.select("Inbox", readonly=True)
+			response, message = self.imap.uid('search', None, self.settings.email_sync_rule)
```

Comment:
unfortunately searching by uid is not supported by all email servers and so cannot be used (some servers will just return all uid's or varied results based of frequency of requests